gakido/backoff.py

```
@@ -0,0 +1,142 @@
+
+    def decorator(func: Callable) -> Callable:
+        def wrapper(*args, **kwargs):
+            attempt = 0
+            last_exc = None
+
+            while attempt < max_attempts:
+                try:
+                    result = func(*args, **kwargs)
+                    # Check for retryable HTTP status codes
+                    if hasattr(result, "status_code") and result.status_code in retryable_status_codes:
+                        raise RetryError(f"Retryable status code: {result.status_code}")
```

Ruff (0.14.13)

Abstract `raise` to an inner function

(TRY301)